src/de.rs

```diff
@@ -1220,39 +1268,69 @@ impl<'de> de::Deserializer<'de> for &mut DeserializerFromEvents<'de, '_> {
             }
             parse_tag(tag)
         }
+        // TODO: switch to JSON enum semantics for JSON content
+        // Robust impl blocked on https://github.com/serde-rs/serde/pull/2420
+        let is_serde_content =
+            std::any::type_name::<V::Value>() == "serde::__private::de::content::Content";
```

Comment:
The `Content` type is technically public. The `type_name` have no guaranties of the returned string, so it may be worth to check for the type directly:
```suggestion
            std::any::type_name::<V::Value>() == std::any::type_name::<serde::__private::de::Content>();
```
or more straightforward. Yes, that technically use of the private API that can be changed in any time... but the current solution will just silently stop to working if such breaking changes would be made. It seems it would be better to get compile-time error instead of hidden runtime errors.

Reply:
I haven't tried what Mingun suggested, but if it works, could you implement that. Could you also move the is_serde_content in the matching arm. We don't need to compare for other type, and since you don't use arm conditions like `Foo if condition => ...` (which I don't have strong opinon about), I think you can move the `is_serde_content` into the arm.